### PR TITLE
ES2015ify and use promises

### PR DESCRIPTION
**Pull request by kevva**
Wish we could do without nesting this much but since we need `fd` all the way down it's the best way to do it.

### CODE REVIEW

**Comment by SamVerschueren** [comment]
Storing `fd` in a temporary variable might be a cleaner approach? It looks like callback-hell :D

File: test.js [hunk 1]
```diff
@@ -1,29 +1,18 @@
...
+import m from '.';
 
-describe('readChunk()', function () {
-	it('should read chunks from a file', function (cb) {
-		readChunk('fixture', 1, 4, function (err, buf) {
-			assert.equal(buf.toString(), 'ello');
-			cb();
-		});
-	});
+test('should read chunks from a file', async t => {
```
Comment (SamVerschueren) [inline test.js]: Remove all the `should` words

**Comment by kevva** [comment]
Thought of that as well, but I rather return it than storing it :P.

File: test.js [hunk 2]
```diff
@@ -1,29 +1,18 @@
-'use strict';
-var assert = require('assert');
-var readChunk = require('./');
+import test from 'ava';
+import m from '.';
```
Comment (sindresorhus) [inline test.js]: Use `./`. See: https://github.com/sindresorhus/eslint-plugin-xo/issues/9#issuecomment-211760642

File: index.js [hunk 3]
```diff
@@ -1,41 +1,31 @@
...
-					return;
-				}
+const fs = require('fs');
+const pify = require('pify');
+
+module.exports = (filepath, pos, len) => {
+	const buf = new Buffer(len);
+
+	return pify(fs.open)(filepath, 'r').then(fd => {
+		return pify(fs.read, {multiArgs: true})(fd, buf, 0, len, pos).then(res => {
```
Comment (kevva) [inline index.js]: If we store `fd` outside, we also need to store `res` outside because we're doing `fs,close` before actually doing anything with `res`.

**Comment by sindresorhus** [comment]
Less nesting please. 

Could do something like this (untested):

```js
module.exports = (filepath, pos, len) => {
	const buf = new Buffer(len);

	let fd;

	return pify(fs.open)(filepath, 'r')
		.then(res => {
			fd = res;
		})
		.then(() => pify(fs.read, {multiArgs: true})(fd, buf, 0, len, pos))
		.then(res => pify(fs.close)(fd).then(() => res))
		.then(res => {
			const bytesRead = res[0];
			let buf = res[1];

			if (bytesRead < len) {
				buf = buf.slice(0, bytesRead);
			}

			return buf;
		});
};
```

**Comment by kevva** [comment]
That works :).

**Comment by sindresorhus** [comment]
Looks good. Thanks :)

**Pull request merged**